python/cugraph/structure/graph_implementation/simpleGraph.py

```diff
@@ -0,0 +1,838 @@
+        Parameters
+        ----------
+        vertex_subset : cudf.Series or iterable container, optional
+            A container of vertices for displaying corresponding degree. If not
+            set, degrees are computed for the entire set of vertices.
+        Returns
+        -------
+        df : cudf.DataFrame
+            GPU DataFrame of size N (the default) or the size of the given
+            vertices (vertex_subset) containing the degrees. The ordering is
+            relative to the adjacency list, or that given by the specified
+            vertex_subset.
+            df['vertex'] : cudf.Series
+                The vertex IDs (will be identical to vertex_subset if
+                specified).
+            df['in_degree'] : cudf.Series
+                The in-degree of the vertex.
+            df['out_degree'] : cudf.Series
+                The out-degree of the vertex.
+        Examples
+        --------
+        >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
+        >>>                   dtype=['int32', 'int32', 'float32'], header=None)
+        >>> G = cugraph.Graph()
+        >>> G.from_cudf_edgelist(M, '0', '1')
+        >>> df = G.degrees([0,9,12])
+        """
+        (
+            vertex_col,
+            in_degree_col,
+            out_degree_col,
+        ) = graph_primtypes_wrapper._degrees(self)
+
+        df = cudf.DataFrame()
+        df["vertex"] = vertex_col
+        df["in_degree"] = in_degree_col
+        df["out_degree"] = out_degree_col
+
+        if self.properties.renumbered is True:
+            df = self.renumber_map.unrenumber(df, "vertex")
+
+        if vertex_subset is not None:
+            df = df[df['vertex'].isin(vertex_subset)]
+
+        return df
+
+    def _degree(self, vertex_subset, x=0):
+        vertex_col, degree_col = graph_primtypes_wrapper._degree(self, x)
+        df = cudf.DataFrame()
+        df["vertex"] = vertex_col
+        df["degree"] = degree_col
+
+        if self.properties.renumbered is True:
+            df = self.renumber_map.unrenumber(df, "vertex")
+
+        if vertex_subset is not None:
+            df = df[df['vertex'].isin(vertex_subset)]
+
+        return df
+
+    def to_directed(self, DiG):
+        """
+        Return a directed representation of the graph.
+        This function sets the type of graph as DiGraph() and returns the
+        directed view.
+        Returns
+        -------
+        G : DiGraph
+            A directed graph with the same nodes, and each edge (u,v,weights)
+            replaced by two directed edges (u,v,weights) and (v,u,weights).
+        Examples
+        --------
+        >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
+        >>>                   dtype=['int32', 'int32', 'float32'], header=None)
+        >>> G = cugraph.Graph()
+        >>> G.from_cudf_edgelist(M, '0', '1')
+        >>> DiG = G.to_directed()
+        """
+        DiG.properties.renumbered = self.properties.renumbered
```

Comment:
Seems like this should at least assert that the incoming object is a compatible Graph type (eg. `isinstance(DiG, Graph)`) before assigning to it.

Reply:
Since DiG comes from the external class where its initialized as type(self) which is the type of object/Graph passed with directed=True, it is ensured that it is compatible